### DISCUSSION TITLE
using --upgrade, whenever it pulls down a new version it deletes my .bat files

### DISCUSSION THREAD

**Discussion by crazygonuts** [vladmandic/sdnext#570] [Q&A]
I created some .bat files to set different command line args before calling webui.bat, but if I use the --upgrade flag and it finds a new version, it deletes my .bat files out of the directory when it upgrades.  I'm pretty new to git, is this normal behavior and I should just put them in another directory or does this sound like a bug?

**Comment by mykeehu** [comment]
No, this is not a bug, it should be put on the exception list. I got the answer here:
https://github.com/vladmandic/automatic/issues/499#issuecomment-1522322215

**Reply by anothertal3** [reply]
In my opinion, this works "by accident":
- If you change the file yourself there might be issues in the future as `.gitignore` is part of the repository and managed by the owner. It could be subject to change at any time.
- If you don't change it you're forced to reuse the names of excluded files which - again - might change at any time.

However, it's a slightly better idea than mine which includes having custom branches (requiring merging and switching after each upgrade) so I will use it as well for now ;)

**Comment by toyxyz** [comment]
I solved it by putting the .bat file in another folder. Of course you need to enter the correct path.

**Comment by vladmandic** [comment]
since only two shell scripts that should not be modified are `webui.bat` and `webui.sh`, i'll change `.gitignore` to:

```log
/*.bat
/*.sh
!webui.bat
!webui.sh
```

which means you can create any bat or sh file and they will not be touched during upgrade except those two.

**Comment by econundrum** [comment]
Personally I think it's easiest to create desktop shortcuts to webui.bat and put your command line arguments in the shortcut.

**Reply by vladmandic** [reply]
i totally agree. but some users prefer alternatives and this is easy fix for me.

**Reply by anothertal3** [reply]
I only use the batch file to activate the venv beforehand. I'm not sure if there's an easy way to achieve that with a link and command line arguments only?

**Reply by vladmandic** [reply]
That's pretty much all thar WebUI.bat/sh do nowadays, all other functionality has been moved to setup.py. Launch.py exists mainly for compatibility reasons as its used by extensions to install themselves.

**Reply by anothertal3** [reply]
Ouch, I must have been positively blind. I've recreated something which was already available.

You're right then: at least for me there's no need for batch files then :)

**Comment by lzardy** [comment]
Just adding here that all other extraneous files are also deleted. This does not include zip files or folders.
Don't be like me and get your blender, LoRA, etc things deleted.

**Reply by vladmandic** [reply]
@lzardy give me one example of such file deleted?

**Reply by lzardy** [reply]
- Folder "Extras" in root folder
- Place ".blender", ".py", ".png", ".jpg" files in it.
- Run "python launch.py --upgrade" in root folder to upgrade.
- Upon message "Upgraded to version: ...", files with extensions listed above will be deleted.

**Reply by lzardy** [reply]
Will note that AUTO111 did not do this and I recently switched to this UI expecting the same, although I understand now that your process is different.

**Comment by vladmandic** [comment]
i've added detailed notes #667 and i'm open to suggestions, but please read first as there are pros and cons.
